CDCgov/prime-reportstream
# Fix delivery & submission history filtering inconsistencies

This PR fixes inconsistent filtering behavior in the delivery and submission history endpoints.

Test Steps:
1. Ensure smoke test data is present in the database
2. Run the backend
3. `GET` the following URLs in your tool of choice and verify the result

| curl | Result |
| ---- | ------ |
| `/api/waters/org/ignore/submissions` | submission history for all senders |
| `/api/waters/org/ignore.ignore-full-elr/submissions` | 🆕 submission history for `ignore-full-elr` senders only |
| `/api/waters/org/ignore.sdfljkd/submissions` | 404 |
| `/api/waters/org/ignore/deliveries` | 🆕 delivery history for all receivers |
| `/api/waters/org/ignore.CSV/deliveries` | delivery history for `CSV` receivers |
| `/api/waters/org/ignore.sdfljkd/deliveries` | 404 |

## Changes

### Old Behavior:
- Submission histories for all sending services was being returned no matter what.
  - e.g. `GET waters/someorg.somesender/submissions` would return all senders for someorg 
- Delivery histories for all receiving services could not be queried (only a specific service)
  - e.g. `GET waters/someorg/deliveries` resulted in 404 for a valid org

### New Behavior (also Release Notes):
- Submissions history can now be retrieved for a specific sender
  - e.g. `GET waters/someorg.somesender/submissions` returns only submissions from `somesender`
- Delivery histories for all receiving services can now be queried
  - e.g. `GET waters/someorg/deliveries` returns delivery history for all receivers of `someorg`
 
## Checklist

### Testing
- [x] Added tests for submissions
- [ ] Added tests for deliveries

### Process
- [x] ~~Updated~~ Drafted the release notes?

## Linked Issues
- Fixes #8019 
- Fixes #8322 

## TODO

- New: #8424
